src/coreComponents/codingUtilities/Section.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Section.cpp
+ */
+
+#include "Section.hpp"
+#include <algorithm>
+
+namespace geos
+{
+
+Section::Section():
+  m_rowMinWidth( 20 )
+{}
+
+void Section::setName( string_view title )
+{
+  m_sectionTitle = title;
+}
+
+void Section::addDescription( string const & description )
+{
+  m_descriptions.push_back( description );
+}
+
+void Section::setMinWidth( integer const & minWidth )
+{
+  m_rowMinWidth = minWidth;
+}
+
+void Section::computeMaxRowSize( string_view title,
+                                 std::vector< string > const & rowsDescription )
+{
+  std::string const footerStr = "End : ";
+  integer const titleLength = footerStr.length() + title.length() + m_marginBorder * 2 + m_nbSpecialChar * 2;
+
+  m_rowLength = std::max( m_rowMinWidth, titleLength );
+
+  if( rowsDescription.size() == 0 )
+  {
+    return;
+  }
+
+  auto it = std::max_element( rowsDescription.begin(),
+                              rowsDescription.end(),
+                              []( auto const & a, auto const & b ) {
+    return a.size() < b.size();
+  } );
```

Comment:
here you can write it, as you prefer:
```suggestion
  auto it = std::max_element( rowsDescription.begin(), rowsDescription.end(),
                              []( auto const & a, auto const & b ) { return a.size() < b.size(); } );
```

Reply:
2ns one better agree